contrib/epee/include/misc_language.h

```diff
@@ -128,6 +128,59 @@ namespace misc_utils
 
   }
 
+  template<typename T>
+  T factorial(T n, T s = 0)
+  {
+    static_assert(std::is_integral<T>::value, "Integral type required.");
+
+    if (s > n || n < 0 || s < 0)
+      return 0;
+
+    // n! / s!
+    // s as in 'stop'
+    T r{1};
+    while (n > s)
+    { 
+      if ((std::numeric_limits<T>::max() / r) < n)
+        return 0;
+
+      r *= n;
+      n--;
+    }
+
+    return r;
+  }
+
+  template<typename T>
+  T n_choose_k(T n, T k)
+  {
+    static_assert(std::is_integral<T>::value, "Integral type required.");
+
+    // n choose k = n! / (k! * (n - k)!)
+    if (n < 0 || k < 0)
+      return 0;
+    else if (n < k)
+      return 1;
```

Comment:
Why `return 1;` here? This is basically another error case as choose k elements from n only makes sense if this statement is `true`, but why differentiate this instead of using `0` like all the other error cases?

Reply:
I changed my mind, `n < k` is also an error case (will update later).